qa/workunits/cephadm/test_cephadm.sh

```diff
@@ -250,11 +250,33 @@ loop_dev=$($SUDO losetup -f)
 $SUDO vgremove -f $OSD_VG_NAME || true
 $SUDO losetup $loop_dev $TMPDIR/$OSD_IMAGE_NAME
 $SUDO pvcreate $loop_dev && $SUDO vgcreate $OSD_VG_NAME $loop_dev
+
+# osd boostrap keyring
+$CEPHADM shell --fsid $FSID --config $CONFIG --keyring $KEYRING -- \
+      ceph auth get client.bootstrap-osd > $TMPDIR/keyring.bootstrap.osd
+
 for id in `seq 0 $((--OSD_TO_CREATE))`; do
     $SUDO lvcreate -l $((100/$OSD_TO_CREATE))%VG -n $OSD_LV_NAME.$id $OSD_VG_NAME
-    $CEPHADM shell --fsid $FSID --config $CONFIG --keyring $KEYRING -- \
-            ceph orch osd create \
-                $(hostname):/dev/$OSD_VG_NAME/$OSD_LV_NAME.$id
+    device_name=/dev/$OSD_VG_NAME/$OSD_LV_NAME.$id
+
+    # prepare the osd
+    $CEPHADM ceph-volume --config $CONFIG --keyring $TMPDIR/keyring.bootstrap.osd -- \
+            lvm prepare --bluestore --data $device_name --no-systemd
+    $CEPHADM ceph-volume --config $CONFIG --keyring $TMPDIR/keyring.bootstrap.osd -- \
+            lvm batch --no-auto $device_name --yes --no-systemd
+
+    # osd id and osd fsid
+    $CEPHADM ceph-volume --config $CONFIG --keyring $TMPDIR/keyring.bootstrap.osd -- \
+            lvm list --format json $device_name > $TMPDIR/osd.map
+    osd_id=$($SUDO cat $TMPDIR/osd.map | jq -cr '.. | ."ceph.osd_id"? | select(.)')
+    osd_fsid=$($SUDO cat $TMPDIR/osd.map | jq -cr '.. | ."ceph.osd_fsid"? | select(.)')
+
+    # deploy the osd
+    $CEPHADM deploy --name osd.$osd_id \
+          --fsid $FSID \
+          --keyring $TMPDIR/keyring.bootstrap.osd \
+          --config $CONFIG \
+          --osd-fsid $osd_fsid
```

Comment:
why? Is the current code broken in some way? this reduces the coverage in mgr/cephadm

Reply:
`ceph orch osd create` was broken in the `mgr/cephadm` layer  ... shouldn't these tests only focus on `bin/cephadm` ?

Reply:
hm. do you have a traceback some something similar?

Reply:
```
+ sudo /tmp/tmp.ZlyRDRVwla/cephadm --image docker.io/ceph/daemon-base:latest-master-devel shell --fsid 00000000-0000-0000-0000-0000deadbeef --config /tmp/tmp.CJGxN6SA7u/tmp.r9LWfuuEb5 --keyring /tmp/tmp.CJGxN6SA7u/tmp.WY52Ho5uw3 -- ceph orch osd create host1:/dev/test_cephadm/test_cephadm.0
Error EINVAL: Traceback (most recent call last):
  File "/usr/share/ceph/mgr/mgr_module.py", line 1070, in _handle_command
    return self.handle_command(inbuf, cmd)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 191, in handle_command
    return dispatch[cmd['prefix']].call(self, cmd, inbuf)
  File "/usr/share/ceph/mgr/mgr_module.py", line 309, in call
    return self.func(mgr, **kwargs)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 144, in wrapper
    return func(*args, **kwargs)
  File "/usr/share/ceph/mgr/orchestrator/module.py", line 458, in _create_osd
    raise_if_exception(completion)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 701, in raise_if_exception
    raise e
TypeError: %d format: a number is required, not str
```

Reply:
mgr logs should show more!